table_helper.cc

```diff
@@ -31,7 +32,22 @@ future<> table_helper::setup_table() const {
     auto& db = qp.db();
 
     if (db.has_schema(_keyspace, _name)) {
-        return make_ready_future<>();
+        if (_alter_cql.empty()) {
+            return make_ready_future<>();
+        } else {
+            auto parsed = cql3::query_processor::parse_statement(_alter_cql);
+            cql3::statements::raw::cf_statement* parsed_cf_stmt = static_cast<cql3::statements::raw::cf_statement*>(parsed.get());
+            ::shared_ptr<cql3::statements::alter_table_statement> statement =
+                            static_pointer_cast<cql3::statements::alter_table_statement>(
+                                            parsed_cf_stmt->prepare(db, qp.get_cql_stats())->statement);
+
+            // Instead of checking if the alteration is possible, we try it, and in case it fails we change nothing
```

Comment:
Why is it ok to just fail and not retry? What are the reasons the alternation could be impossible? Do we have a guarantee that eventually the cluster will move to the new schema that contains username?

Reply:
The reason for the fail is that the alteration has already been done, which could have happened in the same or a different node and I can't actually see any other reason - in the previous version we assumed that the schema is up-to-date here, so in this version it should be in the old state or in a state after altering the old state. (We could manually destroy the table but I don't think we're implementing repairing of such damage)
The cluster should move to the new schema after the first call(or concurrent calls) of `statement->announce_migration`, which is called (indirectly) as early as during the init phase.
Note that this assumes that we're upgrading by adding new columns - current version wouldn't work well with dropping or renaming.
Please correct me if I missed anything.

Reply:
After researching a bit more it looks like all nodes will get the final schema after the last node that didn't get the schema from gossip sends its version with latest timestamp to all nodes.(It probably shouldn't happen in practice - every node aside from the first should get the scheme on joining the cluster)